src/Java.Interop.Tools.Maven/Models/MavenVersion.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using Java.Interop.Tools.Maven.Extensions;
+
+namespace Java.Interop.Tools.Maven.Models;
+
+// https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN8855
+public class MavenVersion : IComparable, IComparable<MavenVersion>, IEquatable<MavenVersion>
+{
+	public string? Major { get; private set; }
+	public string? Minor { get; private set; }
+	public string? Patch { get; private set; }
+	public string RawVersion { get; private set; }
+	public bool IsValid { get; private set; } = true;
+
+	MavenVersion (string rawVersion) => RawVersion = rawVersion;
+
+	public static MavenVersion Parse (string version)
```

Comment:
Should this mirror `Artifact` and have a `TryParse()` method as well?

Reply:
…except it can't fail…. Weird semantics, especially around `.IsValid`…

Reply:
Yeah, it is weird.  There doesn't seem to be any validation on Maven version numbers, users can't technically put anything they want.  If it isn't a "valid" Maven version, it just gets treated like an opaque string for sorting and version matching purposed.

> If you do not follow Maven versioning standards in your project versioning scheme, then for version comparison, Maven interprets the entire version as a simple string.

From https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN8855.